ReText/webenginepreview.py

```diff
@@ -37,12 +38,40 @@ def interceptRequest(self, info):
             info.block(True)
 
 
+class UrlPopup(QLabel):
+
+    def __init__(self, window):
+        super().__init__(window)
+        self.window = window
+
+        self.setStyleSheet('''
+            border: 1px solid #64323232;
+            border-radius: 3px;
+            background: #FAFAFAFA;
+        ''')
+        self.fontHeight = self.fontMetrics().height()
+        self.setVisible(False)
+
+    def pop(self, url):
+        if url:
+            self.setText(url)
+            windowBottom = self.window.rect().bottom()
+            textWidth = self.fontMetrics().horizontalAdvance(url)
+            self.setGeometry(-2, windowBottom-self.fontHeight-5,
+                    textWidth+10, self.fontHeight+10)
+            self.setVisible(True)
+        else:
+            self.setVisible(False)
+
+
 class ReTextWebEnginePage(QWebEnginePage):
     def __init__(self, parent, tab):
         QWebEnginePage.__init__(self, parent)
         self.tab = tab
         self.interceptor = ReTextWebEngineUrlRequestInterceptor(self)
         self.setUrlRequestInterceptor(self.interceptor)
+        self.urlPopup = UrlPopup(self.tab.p)
+        self.linkHovered.connect(self.onLinkHover)
```

Comment:
I think you can do this:
```suggestion
        self.linkHovered.connect(self.urlPopup.pop)
```

and remove the `onLinkHover` method.